pasta_eln/backend.py

```diff
@@ -492,10 +491,11 @@ def useExtractors(self, filePath:Path, shasum:str, doc:dict[str,Any]) -> None:
           for item in doc[meta]:
             if isinstance(doc[meta][item], tuple):
               doc[meta][item] = list(doc[meta][item])
-            if not isinstance(doc[meta][item], (str, int, float, list)) and \
-                    doc[meta][item] is not None:
-              print(' -> simplify ',meta,item, doc[meta][item])
+            try:
+              _ = json.dumps(doc[meta][item])
+            except:
               doc[meta][item] = str(doc[meta][item])
```

Comment:
Are you trying to convert the JSON content within doc[meta][item]  to string here? So only when json dump fails you do the stringify? Why json dumps has to fail to do so and what happens when the json dumps succeeds?

Reply:
@jmurugan-fzj  The underlying issue is that couchDB can only save any content that it can create a json from. Sometimes PASTA gets from the extractor (which the user can change, we cannot ensure that it can be json-fied) content that cannot be converted into a straight json. For instance PIL package has image metadata that uses the PIL classes, and cannot be jsonfied. If that is the case, if one subproperty cannot be jsonfied, then create  a string (which is always possible, and that string is good to go. That leads to a dpi=[500,500] (500 being some PIL class) to become a dpi="[500,500]" and not numbers, but that is better than a crash.